libcxx/test/support/constexpr_random.h

```diff
@@ -0,0 +1,705 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TEST_SUPPORT_CONSTEXPR_RANDOM_H
+#define TEST_SUPPORT_CONSTEXPR_RANDOM_H
```

Comment:
I feel this is quite a lot of code. I know we've had bugs in libc++ with Schrage's algorithm. So I'm a bit reluctant to add this code without any tests; probably in a separate commit. I still wonder whether we shouldn't add a simplified random algorithm. Not only to keep the code simple, but also to keep the number of steps during constant evaluation lower. WDYT?

Reply:
Agreed. I'll look for a simpler RNG.

Reply:
Thanks. It's still a bit more code than I hoped, but I don't think that can be avoided.